Split/Storage/Impressions/ImpressionDao.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  ImpressionDao.swift
+//  Split
+//
+//  Created by Javier L. Avrudsky on 12/11/2020.
+//  Copyright © 2020 Split. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+protocol ImpressionDao {
+    func insert(_ impression: Impression)
+    func getBy(createdAt: Int64, status: Int32, maxRows: Int) -> [Impression]
+    func update(ids: [String], newStatus: Int32)
+    func delete(_ impressions: [Impression])
+}
+
+class CoreDataImpressionDao: ImpressionDao {
+
+    let coreDataHelper: CoreDataHelper
+
+    init(coreDataHelper: CoreDataHelper) {
+        self.coreDataHelper = coreDataHelper
+    }
+
+    func insert(_ impression: Impression) {
+        if let obj = coreDataHelper.create(entity: .impression) as? ImpressionEntity {
+            do {
+                obj.storageId = coreDataHelper.generateId()
+                obj.body = try Json.encodeToJson(impression)
+                obj.createdAt = Date().unixTimestamp()
+                obj.status = StorageRecordStatus.active
```

Comment:
the `ImpressionEntity` has also a testName field. Check Javi if we need to assign a value to it here.

Reply:
Good catch doc! Fixed